server/src/frontend_api.rs

```diff
@@ -1,3 +1,169 @@
+    let context = context.into_inner();
+
+    let evaluated_features = resolve_frontend_features(client_features, context);
+
+    Ok(Json(FrontendResult {
+        toggles: evaluated_features,
+    }))
+}
+
+#[actix_web::post("/proxy/all")]
+async fn post_frontend_features(
+    edge_token: EdgeToken,
+    features_source: web::Data<dyn EdgeProvider>,
+    context: web::Json<Context>,
+) -> EdgeJsonResult<FrontendResult> {
+    let client_features = features_source.get_client_features(edge_token);
+    let context = context.into_inner();
+
+    let evaluated_features = resolve_frontend_features(client_features, context);
+
+    Ok(Json(FrontendResult {
+        toggles: evaluated_features,
+    }))
+}
+
+fn resolve_frontend_features(
+    client_features: ClientFeatures,
+    context: Context,
+) -> Vec<EvaluatedToggle> {
+    let mut engine = EngineState::default();
+    engine.take_state(client_features.clone());
+
+    client_features
+        .features
+        .into_iter()
+        .map(|toggle| {
+            let variant = engine.get_variant(toggle.name.clone(), &context);
+            EvaluatedToggle {
+                name: toggle.name.clone(),
+                enabled: engine.is_enabled(toggle.name, &context),
+                variant: EvaluatedVariant {
+                    name: variant.name,
+                    enabled: variant.enabled,
+                    payload: variant.payload.map(|succ| Payload {
+                        payload_type: succ.payload_type,
+                        value: succ.value,
+                    }),
+                },
+                impression_data: false,
+            }
+        })
+        .collect()
+}
+
+pub fn configure_frontend_api(cfg: &mut web::ServiceConfig) {
+    cfg.service(get_frontend_features);
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use crate::types::{EdgeProvider, FeaturesProvider, TokenProvider};
+    use actix_web::{
+        http::header::ContentType,
+        test,
+        web::{self, Bytes},
+        App,
+    };
+    use serde_json::json;
+    use unleash_types::{
+        client_features::{ClientFeature, ClientFeatures, Variant},
+        frontend::{EvaluatedToggle, EvaluatedVariant, FrontendResult},
+    };
+
+    #[derive(Clone)]
+    struct MockDataSource {}
+
+    impl FeaturesProvider for MockDataSource {
+        fn get_client_features(&self, _token: crate::types::EdgeToken) -> ClientFeatures {
+            return ClientFeatures {
+                version: 1,
+                features: vec![ClientFeature {
+                    name: "test".into(),
+                    enabled: true,
+                    ..ClientFeature::default()
+                }],
+                segments: None,
+                query: None,
+            };
+        }
+    }
+
+    impl TokenProvider for MockDataSource {
+        fn get_known_tokens(&self) -> Vec<crate::types::EdgeToken> {
+            todo!()
```

Comment:
So, sort of like implementing an interface on our mocks on Unleash I assume. We satisfy the TS compiler by implementing the methods but they can just do w/e because it doesn't matter.

Reply:
Yeah, sort of. This is Actix playing silly buggers here, it can't dynamically resolve the types correctly so we gotta nudge it a little. Sadly not a compile error, this results in a runtime error since Actix doesn't know how to actually call the relevant functions